Classes/Hooks/Frontend/TSGenerator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the package bk2k/bootstrap-package.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace BK2K\BootstrapPackage\Hooks\Frontend;
+
+use BK2K\BootstrapPackage\Utility\LanguageUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
+
+/**
+ * TSGenerator
+ */
+class TSGenerator
+{
+    const SYS_LANGUAGE_UID_PLACEHOLDER = '###SYSLANGUAGEUIDPLACEHOLDER###';
+    const LANGUAGE_PLACEHOLDER = '###LANGUAGEPLACEHOLDER###';
+    const LOCALE_PLACEHOLDER = '###LOCALEPLACEHOLDER###';
+    const HREF_LANG_PLACEHOLDER = '###HREFLANGPLACEHOLDER###';
+    const DIRECTION_PLACEHOLDER = '###DIRECTIONPLACEHOLDER###';
+
+    /**
+     * @var string
+     */
+    protected $tempDirectory = 'typo3temp/assets/bootstrappackage/';
```

Comment:
still unsure about the location here. the assets folder is not protected because we will have public data here, the language configuration is private stuff and should never be available anywhere publicly.

Reply:
any suggestion?

Reply:
what about typo3temp/var/Cache/Code/bootstrappackage/?

Reply:
not sure about that one, but i know that there will be changes in the future to that path. see https://github.com/TYPO3/TYPO3.CMS/commit/6361660f70e6ee0398e5b68ae472633415c77103

Reply:
as far as i see this should not break my impl (which i realy don't like but i din't find an other way so far) other place coud be some where in ext:bootstrap_package/Resources/Private, also not nice but as far the file will be recreated should not make troubles...